### ISSUE TITLE
web-ext lint errors when using dynamic import syntax

### ISSUE THREAD

**Issue by pdehaan** [mozilla/web-ext#1770]
### Is this a feature request or a bug?

Feature request maybe?

### What is the current behavior?

I'm trying to run <kbd>npx web-ext lint -s src --self-hosted --ignore-files="vendor/*"</kbd> on https://github.com/mozilla/secure-proxy/ and am getting a few errors.

This is my current output:

```sh
npx web-ext lint -s src --self-hosted --ignore-files="vendor/*"
Applying config files: ./package.json, ./web-ext-config.js
Validation Summary:

errors          3
notices         0
warnings        31

ERRORS:

Code              Message                   Description                                                                                 File                 Line   Column
JS_SYNTAX_ERROR   JavaScript syntax error   There is a JavaScript syntax error in your code; validation cannot continue on this file.   background/main.js   1      1
JS_SYNTAX_ERROR   JavaScript syntax error   There is a JavaScript syntax error in your code; validation cannot continue on this file.   popup/popup.js       4      24
JS_SYNTAX_ERROR   JavaScript syntax error   There is a JavaScript syntax error in your code; validation cannot continue on this file.   popup/view.js        7      1
WARNINGS:

Code                    Message                                       Description                                                                                     File                             Line   Column
UNSUPPORTED_API         experiments.proxyutils is not supported       This API has not been implemented by Firefox.                                                   background/connectivity.js       8      5
UNSUPPORTED_API         experiments.proxyutils is not supported       This API has not been implemented by Firefox.                                                   background/connectivity.js       13     5
UNSUPPORTED_API         experiments.proxyutils is not supported       This API has not been implemented by Firefox.                                                   background/fxa.js                52     25
UNSUPPORTED_API         experiments.proxyutils is not supported       This API has not been implemented by Firefox.                                                   background/network.js            69     15
UNSUPPORTED_API         captivePortal.canonicalURL is not supported   This API has not been implemented by Firefox.                                                   background/network.js            93     9
UNSUPPORTED_API         captivePortal.canonicalURL is not supported   This API has not been implemented by Firefox.                                                   background/network.js            94     33
UNSUPPORTED_API         experiments.proxyutils is not supported       This API has not been implemented by Firefox.                                                   background/network.js            96     27
UNSUPPORTED_API         experiments.proxyutils is not supported       This API has not been implemented by Firefox.                                                   background/network.js            402    5
UNSUPPORTED_API         browserSettings.ftpProtocolEnabled is not     This API has not been implemented by Firefox.                                                   background/network.js            414    9
                        supported
UNSUPPORTED_API         browserSettings.ftpProtocolEnabled is not     This API has not been implemented by Firefox.                                                   background/network.js            416    7
                        supported
UNSUPPORTED_API         experiments.proxyutils is not supported       This API has not been implemented by Firefox.                                                   background/network.js            419    7
UNSUPPORTED_API         experiments.proxyutils is not supported       This API has not been implemented by Firefox.                                                   background/network.js            433    5
UNSUPPORTED_API         experiments.proxyutils is not supported       This API has not been implemented by Firefox.                                                   background/network.js            435    5
UNSUPPORTED_API         experiments.proxyutils is not supported       This API has not been implemented by Firefox.                                                   background/network.js            440    5
UNSUPPORTED_API         experiments.proxyutils is not supported       This API has not been implemented by Firefox.                                                   background/network.js            442    5
UNSUPPORTED_API         experiments.proxyutils is not supported       This API has not been implemented by Firefox.                                                   background/network.js            455    5
UNSUPPORTED_API         experiments.proxyutils is not supported       This API has not been implemented by Firefox.                                                   background/network.js            457    5
UNSUPPORTED_API         captivePortal.canonicalURL is not supported   This API has not been implemented by Firefox.                                                   background/proxyDownChecker.js   16     9
UNSUPPORTED_API         captivePortal.canonicalURL is not supported   This API has not been implemented by Firefox.                                                   background/proxyDownChecker.js   17     38
UNSUPPORTED_API         experiments.proxyutils is not supported       This API has not been implemented by Firefox.                                                   background/proxyDownChecker.js   19     27
UNSUPPORTED_API         experiments.proxyutils is not supported       This API has not been implemented by Firefox.                                                   background/ui.js                 47     16
UNSUPPORTED_API         experiments.proxyutils is not supported       This API has not been implemented by Firefox.                                                   background/ui.js                 338    13
UNSUPPORTED_API         experiments.proxyutils is not supported       This API has not been implemented by Firefox.                                                   background/ui.js                 343    11
UNSUPPORTED_API         experiments.proxyutils is not supported       This API has not been implemented by Firefox.                                                   background/ui.js                 356    7
UNSUPPORTED_API         experiments.proxyutils is not supported       This API has not been implemented by Firefox.                                                   background/ui.js                 362    5
UNSUPPORTED_API         experiments.proxyutils is not supported       This API has not been implemented by Firefox.                                                   background/ui.js                 376    7
UNSUPPORTED_API         experiments.proxyutils is not supported       This API has not been implemented by Firefox.                                                   background/ui.js                 383    7
UNSUPPORTED_API         experiments.proxyutils is not supported       This API has not been implemented by Firefox.                                                   background/ui.js                 389    5
UNSUPPORTED_API         experiments.proxyutils is not supported       This API has not been implemented by Firefox.                                                   background/ui.js                 467    30
UNSUPPORTED_API         experiments.proxyutils is not supported       This API has not been implemented by Firefox.                                                   background/ui.js                 493    5
UNSAFE_VAR_ASSIGNMENT   Unsafe assignment to innerHTML                Due to both security and performance concerns, this may not be set using dynamic values which   commons/template.js              60     5
                                                                      have not been adequately sanitized. This can lead to security issues or fairly serious
                                                                      performance degradation.
```

Where the errors in question are from:

1. [background/main.js:1](https://github.com/mozilla/secure-proxy/blob/c8fb3a5c37ad5a378a12fe533160f779b668404f/src/background/main.js#L1)
    ```js
    import {ConnectionTester} from "./connection.js";
    ```

1. [popup/popup.js:4](https://github.com/mozilla/secure-proxy/blob/c8fb3a5c37ad5a378a12fe533160f779b668404f/src/popup/popup.js#L4)
    ```js
    const {View} = await import("./view.js");
    ```

1. [popup/view.js:7](https://github.com/mozilla/secure-proxy/blob/c8fb3a5c37ad5a378a12fe533160f779b668404f/src/popup/view.js#L7)
    ```js
    export class View {
    ```

### What is the expected or desired behavior?

No errors when using ES6+ `import`/`export` syntax.
I tried searching for existing issues to watch, but couldn't find any (but it's a tricky search).

### Version information (for bug reports)

* **Firefox version**:
* **Your OS and version**:
* Paste the output of these commands:

Using macOS 10.14.latest

```sh
node --version && npm --version && npx web-ext --version

v12.11.1
6.11.3
3.2.1
```

---

Also, not sure if there is a way to suppress all the `experiments.* is not supported       This API has not been implemented by Firefox.` warnings.

**Comment by Rob--W** [comment]
Slightly related issue about the syntax errors: #1759 (similar in that special parsers are needed to parse the source files).

I'm not sure if it is currently possible to ignore specific warnings / errors.

**Comment by pdehaan** [comment]
Actually, maybe this belongs in **addons-linter** repo.
I can reproduce w/ <kbd>npx addons-linter --self-hosted ./frontend/public/downloads/browser/v15/secure-proxy.xpi</kbd>

Not sure if somebody can move this to the other repo, or if I should just copy/paste.

**Comment by Rob--W** [comment]
Duplicate the report in addons-linter and link to this issue. I'll then mark this bug as blocked by upstream.

**Comment by pdehaan** [comment]
- [x] https://github.com/mozilla/addons-linter/issues/2940
- [ ] [pending] ability to ignore certain warnings.
- [ ] [pending] ability to ignore specific files (which *web-ext* supports via `--ignore-files="vendor/*"`).

**Comment by Juraj-Masiar** [comment]
I have a same issue here. I'm trying to add Vue.js to my addon but I'm getting error in "vue.esm.browser.min.js" file because of `export default ` at the end.
Plus one more error in the file where I'm dynamically importing it using: 
```
const {default: Vue} = await import(/* webpackIgnore: true */ '../vuejs/vue.esm.browser.min.js');
```

**Is this gonna fail also when releasing my addon on AMO?** Or are these libraries automatically skipped? I think I read somewhere they are skipped, but I cant find it now... there only the short official text that reviewer will check the hash:
https://extensionworkshop.com/documentation/publish/third-party-library-usage/

**Comment by rpl** [comment]
Closing as fixed (by updating addons-linter to v2, #1946) in web-ext v5.0.0 release.
